### ISSUE TITLE
Files with %-sign in names cannot be opened

### ISSUE THREAD

**Issue by fwolfst** [nextcloud/richdocuments#469]
**Describe the bug**
It seems that I cannot open Files where the filename contains a "%"-sign.
The following localized error shows up in the online editor:
"Es konnte kein Verbindung zu Ihrem Dokument hergestellt werden. Bitte versuchen Sie es erneut."
(~Cannot connect to Server. Please try again)

**To Reproduce**
1. Go to "Files"
2. Click on "+" -> "New Document" -> Chosse empty template
3. A half-populated editor shows up (no "paper" can be seen, but much of the rest of the interface)
4. Go back and open the file via click, above error message pops up.

Alternatively, click on "+" and "Upload" to upload a file with a "%"-sign it its filename.

**Expected behavior**
The editor should show the document just fine

**Client details:**
 - OS: Ubuntu 18.04
 - Browser: Firefox 65
 - Device: desktop

## Server details

**Operating system**: Ubuntu 16.04.

**Web server:** Apache

**Database:** MySQL/MariaDB

**PHP version:** 7.2.14-1

**Nextcloud version:** 15.0.5

**Version of the richdocuments app** 3.2.3

**Version of Collabora Online**
LOOLWSD 4.0.1 (git hash: e1b0096)

LOKit Collabora Office 6.0-25 (git hash: 518234a)

1) **How would I update the locale of the collabora docker container (I assume it would not help, but it has been a solution for another issue regarding filenames: https://github.com/nextcloud/richdocuments/issues/204 )**
2) **Where can I find/access the collabora (running in a a container) logs?**
<details>
<summary>Logs</summary>
#### Nextcloud log (data/nextcloud.log)
```
(nothing appears when the error is triggered)
```

#### Browser log
```
Network: A 101 request (switching protocols), afterwards a 200 for the error message.
Console: Nothing different from a working file
```
</details>

**Comment by fwolfst** [comment]
**Edit** Collabora-logs: (via `docker logs -f CONTAINERNAME`):
I replaced stuff in DOMAIN, ACCESSTOKEN and UUID.
```
wsd-00030-22499 2019-04-02 14:29:10.314177 [ docbroker_270 ] ERR  Failed to add session to [/index.php/apps/richdocuments/wopi/files/150444_oc336ypwao1b] with URI [https://DOMAIN/index.php/apps/richdocuments/wopi/files/150444_oc336ypwao1b?access_token=ACCESSTOKEN&access_token_ttl=0&permission=edit]: Bad URI syntax| wsd/DocumentBroker.cpp:1091
wsd-00030-22499 2019-04-02 14:29:10.314318 [ docbroker_270 ] ERR  Error while loading : Bad URI syntax| wsd/LOOLWSD.cpp:2674
wsd-00030-22499 2019-04-02 14:29:10.314668 [ docbroker_270 ] WRN  Child session [1cdf] not found to forward message: load url=https://DOMAIN/index.php/apps/richdocuments/wopi/files/150444_oc336ypwao1b?access_token=ACCESSTOKEN&access_token_ttl=0&permission=edit authorid=dddefb9a-2084-1035-9c19-6f8938c19302 xauthorid=UUID author=Felix%20Wolfsteller xauthor=Felix%20WolfstellerFelix%20Wolfsteller authorextrainfo=%7B%0A%20%20%22avatar%22%20:%20%22https:%5C/%5C/DOMAIN%5C/index.php%5C/avatar%5C/dd...| wsd/DocumentBroker.cpp:1736
kit-22483-00032 2019-04-02 14:29:11.316632 [ loolkit ] WRN  Kit connection lost without exit arriving from wsd. Setting TerminationFlag| kit/Kit.cpp:2218
wsd-00030-22499 2019-04-02 14:29:11.319016 [ docbroker_270 ] ERR  Invalid or unknown session [1cdf] to remove.| wsd/DocumentBroker.cpp:1160
wsd-00030-22499 2019-04-02 14:29:11.319263 [ docbroker_270 ] ERR  No socket associated with WebSocketHandler 0x7f9fc00113a0| ./net/WebSocketHandler.hpp:108
wsd-00030-00031 2019-04-02 14:29:11.319849 [ prisoner_poll ] WRN  Prisoner connection disconnected but without valid socket.| wsd/LOOLWSD.cpp:1776
wsd-00030-22503 2019-04-02 14:29:12.167782 [ docbroker_271 ] ERR  Failed to add session to [/index.php/apps/richdocuments/wopi/files/150444_oc336ypwao1b] with URI [https://DOMAIN/index.php/apps/richdocuments/wopi/files/150444_oc336ypwao1b?access_token=ACCESSTOKEN&access_token_ttl=0&permission=edit]: Bad URI syntax| wsd/DocumentBroker.cpp:1091
wsd-00030-22503 2019-04-02 14:29:12.167932 [ docbroker_271 ] ERR  Error while loading : Bad URI syntax| wsd/LOOLWSD.cpp:2674
wsd-00030-22503 2019-04-02 14:29:12.168247 [ docbroker_271 ] WRN  Child session [1ce0] not found to forward message: load url=https://DOMAIN/index.php/apps/richdocuments/wopi/files/150444_oc336ypwao1b?access_token=ACCESSTOKEN&access_token_ttl=0&permission=edit authorid=dddefb9a-2084-1035-9c19-6f8938c19302 xauthorid=UUID author=Felix%20Wolfsteller xauthor=Felix%20WolfstellerFelix%20Wolfsteller authorextrainfo=%7B%0A%20%20%22avatar%22%20:%20%22https:%5C/%5C/DOMAIN%5C/index.php%5C/avatar%5C/dd...| wsd/DocumentBroker.cpp:1736
wsd-00030-22503 2019-04-02 14:29:13.169729 [ docbroker_271 ] ERR  Invalid or unknown session [1ce0] to remove.| wsd/DocumentBroker.cpp:1160
wsd-00030-22503 2019-04-02 14:29:13.169882 [ docbroker_271 ] ERR  No socket associated with WebSocketHandler 0x7f9fc00113a0| ./net/WebSocketHandler.hpp:108
kit-22500-00032 2019-04-02 14:29:13.169874 [ loolkit ] WRN  Kit connection lost without exit arriving from wsd. Setting TerminationFlag| kit/Kit.cpp:2218
wsd-00030-00031 2019-04-02 14:29:13.170298 [ prisoner_poll ] WRN  Prisoner connection disconnected but without valid socket.| wsd/LOOLWSD.cpp:1776
```

**Comment by fwolfst** [comment]
Maybe a bug in COLLABORA itself:
https://www.collaboraoffice.com/collabora-online-4-release-notes/
```
Bugfixes

    Fix Save As when ‘%’ character is used within the filename
```
Will try out in the following weeks.

**Comment by joergmschulz** [comment]
Indeed. In my installation (4.0.3 (git hash: 81ea0090)), loolwsd crashes on filenames containing a %.

**Comment by juliushaertl** [comment]
cc @thebearon

**Comment by joergmschulz** [comment]
4.1.0 (git hash: 4185361e) does _not_ fix the issue.

**Comment by joergmschulz** [comment]
Possibly, the following info in loolwsd.log is related:
```
wsd-22493-05587 2019-05-15 20:23:19.910459 [ docbroker_006 ] ERR  Failed to add session to [/apps/richdocuments/wopi/files/14718_oczc0pmddo8n] with URI [https://test.blablabla.de/apps/richdocuments/wopi/files/14718_oczc0pmddo8n?access_token=blablabla&
access_token_ttl=0&permission=edit]: Bad URI syntax| wsd/DocumentBroker.cpp:1126
wsd-22493-05587 2019-05-15 20:23:19.910504 [ docbroker_006 ] ERR  Error while loading : Bad URI syntax| wsd/LOOLWSD.cpp:2695
wsd-22493-05587 2019-05-15 20:23:19.910683 [ docbroker_006 ] WRN  Child session [003d] not found to forward message: load url=https://test.sohoserver.de/apps/richdocuments/wopi/files/14718_oczc0pmddo8n?access_token=blablaba&access_token_ttl=0&permission=edit authorid=joerg.schulz xauthorid=joerg.schulz author=joerg.schulz xauthor=joerg.schulz authorextrainfo=%7B%0A%20%20%22avatar%22%20:%20%22https:%5C/%5C/test.blablaba.de%5C/avatar%5C/blablaba readonly=0 lang=de-DE| wsd/DocumentBroker.cpp:1763
```

**Comment by joergmschulz** [comment]
This is not an issue of the richdocuments app but of the libreoffice / collabora app. It will go away with more recent versions of libreoffice. The current alpha versions handle these files correctly.

**Comment by thebearon** [comment]
This has been fixed with Collabora Online / CODE 4.0.5.
